src/mngt/ocf_mngt_cache.c

```diff
@@ -696,7 +701,6 @@ static int _ocf_mngt_init_prepare_cache(struct ocf_cachemng_init_params *param,
 	cache->backfill.max_queue_size = cfg->backfill.max_queue_size;
 	cache->backfill.queue_unblock_size = cfg->backfill.queue_unblock_size;
 
-	env_rwsem_down_write(&cache->lock); /* Lock cache during setup */
```

Comment:
You left "ocf_mngt_cache_unlock" in line 1227 - looks like an error. I'm not sure if this lock during start is needed, I would suggest to leave it though.

Reply:
Cache is automatically locked during `ocf_mngt_cache_lock_init()`, so it's ok. It's needed, because cache object is added to ctx cache list before the start procedure is finished, so another thread iterating over caches in ctx would be able to access incomplete cache if it weren't locked.